ludwig/models/llm.py

```diff
@@ -237,6 +248,7 @@ def forward(
 
             return outputs
 
+        ####SXK why dont we need the calculation of gradients here
```

Comment:
We're computing a forward pass using the model, and we don't need to store the activations since in the generation case, we're not trying to update weights or compute gradients. The torch.no_grad() just freezes the pretrained models weights. Otherwise it's going to generate logits etc which we don't actually need. It also effectively treats the model like it's in "eval" mode